compiler/src/iree/compiler/GlobalOptimization/FuseDequantizationMatmul.cpp

```diff
@@ -219,37 +219,31 @@ struct QuantizedMatmulRewriter {
 // TODO(#) Have stricter matching on inputs. There may be cases where
 // the current matching fails
 std::optional<SmallVector<OpOperand *>>
-QuantizedMatmulRewriter::getDequantMatmulInputs_f32() {
+QuantizedMatmulRewriter::getDequantMatmulInputs() {
   assert(!failed(isContractionWithTwoReductions(matmul)) &&
          "expected `matmul` to be a contraction with two reduction dimensions");
   assert(!failed(isGroupedDequantizationOp(dequant)) &&
          "expected `dequant` to be a grouped dequantization");
   OpOperand *scales, *zps, *quantMat, *unquantMat, *dequantMat;
-  for (int operandIdx = 0; operandIdx < dequant.getNumDpsInputs();
-       operandIdx++) {
-    OpOperand *operand = dequant.getDpsInputOperand(operandIdx);
-    Value input = operand->get();
-    RankedTensorType inputType =
-        llvm::dyn_cast<RankedTensorType>(input.getType());
-    if (!inputType) {
-      continue;
-    }
-    if (inputType.getElementTypeBitWidth() != 32) {
-      quantMat = operand;
-      continue;
-    }
-    for (Operation &bodyOp : dequant.getBlock()->getOperations()) {
-      if (isa<arith::MulFOp>(bodyOp)) {
-        if (bodyOp.getOperand(1) ==
-            dequant.getBlock()->getArgument(operandIdx)) {
-          scales = operand;
-          break;
-        }
-      } else if (isa<arith::SubFOp>(bodyOp)) {
-        if (bodyOp.getOperand(1) ==
-            dequant.getBlock()->getArgument(operandIdx)) {
-          zps = operand;
-          break;
+  auto maps = dequant.getIndexingMapsArray();
+  for (auto map :
+       llvm::enumerate(SmallVector<AffineMap>(maps.begin(), maps.end() - 1))) {
```

Comment:
nit: we can use structured binding, which has better readability.

```suggestion
  for (auto [idx, map] :
       llvm::enumerate(SmallVector<AffineMap>(maps.begin(), maps.end() - 1))) {
```